src/main/java/com/auctionapp/api/service/UserService.java

```diff
@@ -25,30 +36,72 @@ public UserDto getUser(final String email) {
         throw new RuntimeException("User with email " + email + " does not exist!");
 	}
 
+	public UserDto getUserById(final UUID id) {
+		Optional<User> user = userRepository.findById(id);
+        if (user.isPresent()) {
+			return toPayload(user.get());
+		}
+        throw new RuntimeException("User with id " + id + " does not exist!");
+	}
+
+	public void deactivateUser(final UUID id) {
+		Optional<User> user = userRepository.findById(id);
+		user.get().setStatus(Status.INACTIVE);
+		userRepository.save(user.get());
+	}
+
+	public UserDto update(UUID id, UserDto user) {
+        User updatedUser = fromPayload(user);
+        updatedUser = userRepository.save(updatedUser);
+		paymentDetailsRepository.save(updatedUser.getPaymentDetails());
+		shippingDetailsRepository.save(updatedUser.getShippingDetails());
```

Comment:
Is this necessary? Shouldn't the relationships be set up in a way so that if you save the parent, the children also get updated?

Reply:
I forgot to delete it after I added cascade on user entity, I've checked now, everything works without these lines